models/Freelance.js

```diff
@@ -79,6 +79,28 @@ Freelance.pre('save', function (next) {
 	next();
 
 });
+//call of asynchronous, parallel pre
+// method 'validate' is not executed until done() is called.
+// validate is also called for every update, not only save.
+Freelance.pre('validate', true, function(next, done){
+	var count = 0;
+	next();
+	if (this.reviews.length == 0) done();
+	for (var item of this.reviews){
+		Review.findById(item).exec((err, review)=>{
+			if (err) throw err;
+			if (review){
+				this.avgScore += review.score;
+				count++;
+			}
+			if (count >= this.reviews.length) { 
+				this.avgScore = Math.round(this.avgScore/this.reviews.length); // different possibilities for rounding: floor, ceil
```

Comment:
I actually would like to see the scores in the front-end as I suggested above, since for example a 4 could range from 3.75 to 4.24 and over a big number of reviews, it’s quite the difference.
For the box selection rounded or cut away values should be ok.